python/lsst/ts/scheduler/driver.py

```diff
@@ -10,6 +13,80 @@
 __all__ = ["Driver", "DriverParameters"]
 
 
+class DriverTarget(Target):
+    def __init__(self, targetid=0, fieldid=0, band_filter="",
+                 ra_rad=0.0, dec_rad=0.0, ang_rad=0.0,
+                 num_exp=0, exp_times=[]):
+        """Initialize the class.
+
+        Parameters
+        ----------
+        targetid : int
+            A unique identifier for the given target.
+        fieldid : int
+            The ID of the associated OpSim field for the target.
+        band_filter : str
+            The single character name of the associated band filter.
+        ra_rad : float
+            The right ascension (radians) of the target.
+        dec_rad : float
+            The declination (radians) of the target.
+        ang_rad : float
+            The sky angle (radians) of the target.
+        num_exp : int
+            The number of requested exposures for the target.
+        exp_times : list[float]
+            The set of exposure times for the target. Needs to length
+            of num_exp.
+        """
+        super().__init__(targetid=targetid, fieldid=fieldid, band_filter=band_filter, ra_rad=ra_rad,
+                         dec_rad=dec_rad, ang_rad=ang_rad, num_exp=num_exp, exp_times=exp_times)
+
+    def get_script_config(self):
+        """Returns an yaml string with the configuration to be used for the observing script.
+
+        Returns
+        -------
+        config_str: str
+        """
+        script_config = {'targetid': self.targetid,
+                         'band_filter': self.filter,
+                         'ra': self.ra,
+                         'dec': self.dec,
+                         'ang': self.ang,
+                         'num_exp': self.num_exp,
+                         'exp_times': self.exp_times}
+
+        return yaml.safe_dump(script_config)
+
+    def fill_topic(self, topic_target):
+        """ Fill in information on a given SAL topic.
```

Comment:
I suggest "Set data for a target event" (or "...SAL target event").

Is it important to fill an existing event instead of constructing a new one? Returning a new one would make the interface a bit easier to use (and harder to misuse).